lib/eth/abi.rb

```diff
@@ -291,6 +291,17 @@ def decode_primitive_type(type, data)
       end
     end
 
+    # Build event signature string from ABI interface.
+    #
+    # @param interface [Hash] ABI event interface.
+    # @return [String] interface signature string.
+    def signature(interface)
```

Comment:
Generating a signature technically works for functions, errors as well as events. I went ahead and added to the more general module. I didn't have any use case for it here. So if you feel like this should just go into the `Events` module, that's cool with me.